heroic-core/src/main/java/com/spotify/heroic/metric/MetricManagerModule.java

```diff
@@ -519,11 +411,7 @@ public MetricManagerModule build() {
                 fetchParallelism.orElse(DEFAULT_FETCH_PARALLELISM),
                 failOnLimits.orElse(DEFAULT_FAIL_ON_LIMITS),
                 smallQueryThreshold.orElse(DEFAULT_SMALL_QUERY_THRESHOLD),
-                mutateRpcTimeoutMs.orElse(DEFAULT_MUTATE_RPC_TIMEOUT_MS),
-                readRowsRpcTimeoutMs.orElse(DEFAULT_READ_ROWS_RPC_TIMEOUT_MS),
-                shortRpcTimeoutMs.orElse(DEFAULT_SHORT_RPC_TIMEOUT_MS),
-                maxScanTimeoutRetries.orElse(DEFAULT_MAX_SCAN_TIMEOUT_RETRIES),
-                maxElapsedBackoffMs.orElse(DEFAULT_MAX_ELAPSED_BACKOFF_MILLIS)
+                connectionSettings.orElse(MetricsConnectionSettings.createDefault())
```

Comment:
I don't think `connectionSettings` can ever be a null wrapper since it always creates a new object in the constructor. I would change the type to remove the `Optional` and this build method can just pass `connectionSettings` as is.